common/src/main/java/io/pravega/common/util/CompositeByteArraySegment.java

```diff
@@ -217,6 +218,21 @@ public int copyTo(ByteBuffer target) {
         return length;
     }
 
+    @Override
+    public List<ByteBuffer> getContents() {
+        ArrayList<ByteBuffer> result = new ArrayList<>();
+        for (int i = 0; i < this.arrays.length; i++) {
+            byte[] a = getArray(i, false);
```

Comment:
because `byte[][] arrays = new byte[x][y]` will create `y` arrays of `x` elements. This defeats the purpose of this class, which is a view of a sparse array (only allocates sub-arrays as they are needed).